src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs

```diff
@@ -667,5 +667,231 @@ public void SarifTransformerTests_ToVersionOne_OneRunWithInvocationAndNotificati
+                  }
+                ]
+              }
+            }
+          ]
+        }
+      ],
+      ""logicalLocations"": {
+        ""Rules.SecureHashAlgorithmRule.Evaluate"": {
+          ""name"": ""Evaluate"",
+          ""kind"": ""some kind""
+        },
+        ""Rules.SecureHashAlgorithmRule.Register"": {
+          ""name"": ""InvalidName""
+        },
+        ""ExecutionEngine.Engine.FooBar"": {
+          ""name"": ""EvaluateRule"",
+          ""fullyQualifiedName"": ""ExecutionEngine.Engine.EvaluateRule"",
+          ""kind"": ""another kind""
+        },
+        ""Rules.SecureHashAlgorithmRule.Register-0"": {
+          ""name"": ""Register"",
+          ""fullyQualifiedName"": ""Rules.SecureHashAlgorithmRule.Register""
+        },
+        ""ExecutionEngine.Engine.EvaluateRule"": {
+          ""name"": ""EvaluateRule""
+        }
+      },
+      ""results"": []
+    }
+  ]
+}";
+
+            const string V1LogExpectedText =
+@"{
+  ""$schema"": ""http://json.schemastore.org/sarif-1.0.0"",
+  ""version"": ""1.0.0"",
+  ""runs"": [
+    {
+      ""tool"": {
+        ""name"": ""CodeScanner"",
+        ""semanticVersion"": ""2.1.0""
+      },
+      ""invocation"": {},
+      ""logicalLocations"": {
+        ""Rules.SecureHashAlgorithmRule.Evaluate"": {
+          ""name"": ""Evaluate"",
+          ""kind"": ""some kind""
+        },
+        ""Rules.SecureHashAlgorithmRule.Register"": {
+          ""name"": ""InvalidName""
+        },
+        ""ExecutionEngine.Engine.FooBar"": {
+          ""name"": ""EvaluateRule"",
+          ""kind"": ""another kind""
+        },
+        ""Rules.SecureHashAlgorithmRule.Register-0"": {
+          ""name"": ""Register""
+        },
+        ""ExecutionEngine.Engine.EvaluateRule"": {
+          ""name"": ""EvaluateRule""
+        }
+      },
+      ""results"": [],
+      ""toolNotifications"": [
+        {
+          ""id"": ""CTN0001"",
+          ""message"": ""Unhandled exception."",
+          ""level"": ""error"",
+          ""exception"": {
+            ""kind"": ""ExecutionEngine.RuleFailureException"",
+            ""message"": ""Unhandled exception during rule evaluation."",
+            ""stack"": {
+              ""message"": ""This is the stack messasge."",
+              ""frames"": [
+                {
+                  ""message"": ""Exception thrown"",
+                  ""uri"": ""file:///C:/src/main.cs"",
+                  ""line"": 15,
+                  ""column"": 9,
+                  ""module"": ""RuleLibrary"",
+                  ""threadId"": 52,
+                  ""fullyQualifiedLogicalName"": ""Rules.SecureHashAlgorithmRule.Evaluate"",
+                  ""address"": 10092852
+                },
+                {
+                  ""uri"": ""file:///C:/src/main.cs"",
+                  ""module"": ""RuleLibrary"",
+                  ""threadId"": 52,
+                  ""fullyQualifiedLogicalName"": ""Rules.SecureHashAlgorithmRule.Register-0"",
```

Comment:
In v2, the `stackFrame.fullyQualifiedLogicalName` property is always the key into the files dictionary (if the files dictionary exists). So you have to do this:
```
If the files dictionary exists
    AND it has a key that matches v2 stackFrame.fullyQualifiedLogicalName
    AND the file object specified by that key has a fullyQualifiedName property
THEN
    set v1 stackFrame.fullyQualifiedLogicalName to v2 file.fullyQualifiedName
    set v1 stackFrame.logicalLocationKey to v2 stackFrame.fullyQualifiedLogicalName
ELSE
    set v1 stackFrame.fullyQualifiedLogicalName to v2 stackFrame.fullyQualifiedLogicalName
```

Reply:
Fixed. This actually clarifies it quite a bit, thanks